pkg/controller/master/supportbundle/manager.go

```diff
@@ -0,0 +1,207 @@
+package supportbundle
+
+import (
+	"errors"
+	"fmt"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	ctlappsv1 "github.com/rancher/wrangler/pkg/generated/controllers/apps/v1"
+	ctlcorev1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+
+	"github.com/harvester/harvester/pkg/version"
+)
+
+type Manager struct {
+	deployments ctlappsv1.DeploymentClient
+	nodeCache   ctlcorev1.NodeCache
+	services    ctlcorev1.ServiceClient
+}
+
+func (m *Manager) getManagerName(sb *harvesterv1.SupportBundle) string {
+	return fmt.Sprintf("supportbundle-manager-%s", sb.Name)
+}
+
+func (m *Manager) Create(sb *harvesterv1.SupportBundle, image string) error {
+	deployName := m.getManagerName(sb)
+	logrus.Debugf("creating deployment %s with image %s", deployName, image)
+
+	serviceAccountName, err := m.getServiceAccountName(sb.Namespace)
+	if err != nil {
+		return err
+	}
+
+	nodes, err := m.getNodes()
+	if err != nil {
+		return err
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployName,
+			Namespace: sb.Namespace,
+			Labels: map[string]string{
+				"app":                 AppManager,
+				SupportBundleLabelKey: sb.Name,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Name:       sb.Name,
+					Kind:       sb.Kind,
+					UID:        sb.UID,
+					APIVersion: sb.APIVersion,
+				},
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": AppManager},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":                 AppManager,
+						SupportBundleLabelKey: sb.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "manager",
+							Image:           image,
+							Args:            []string{"/usr/bin/support-bundle-utils", "manager"},
+							ImagePullPolicy: corev1.PullAlways,
```

Comment:
Use IfNotPresent policy so that it works in offline environments

Reply:
I also create a setting `SupportBundleImagePullPolicy` for image pull policy because sometimes at debugging field we may want to load another new-modified image.